### PR TITLE
Allow use-defined precision in `XYZWriter`

### LINKED ISSUE #4775: Allow use-defined precision in XYZWriter
## Is your feature request related to a problem? ##
I found a loss of precision when writing XYZ files with XYZWriter (which should be 10.5f), which might be a problem if the absolute values of data are small. 

## Describe the solution you'd like ##
I would like to make the precision as a user-defined variable, as what XTCWriter have implemented. 

## Additional context ##
A possible implementation and the corresponding UT in https://github.com/MDAnalysis/mdanalysis/pull/4771

### PR DESCRIPTION
**Pull request by ChiahsinChu**
Fixes #4775 

Changes made in this Pull Request:
 - add `precision` for `XYZWriter`
 - UT for new feature

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4771.org.readthedocs.build/en/4771/

### CODE REVIEW

**Review by IAlibay — DISMISSED** [review]
Following up from discord, could you please open up an issue to discuss what you are trying to achieve here please? The ASE import cannot happen unfortunately.

**Comment by ChiahsinChu** [comment]
> Following up from discord, could you please open up an issue to discuss what you are trying to achieve here please? The ASE import cannot happen unfortunately.

Issue #4772

**Comment by orbeckst** [comment]
@ChiahsinChu thank you for your contribution. Could you please open an issue that requests the feature that this PR is adding, namely "setting output precision for XYZ Writer"? Before commenting on the implementation in a PR we first need to decide if the requested feature should be added. In the issue provide support in favor of this feature, e.g., documentation for feature specification, description of need etc.

Thank you!

**Comment by orbeckst** [comment]
I am asking for an issue because the issue #4772 that you linked does not seem to have anything to do with the content of this PR. Having separate issues for separate features is a good thing!

**Comment by orbeckst** [comment]
@ChiahsinChu do you want to re-open this PR?

File: package/CHANGELOG [hunk 1]
```diff
@@ -18,7 +18,7 @@ The rules for this file:
          ljwoods2, aditya292002, pstaerk, PicoCentauri, BFedder,
          tyler.je.reddy, SampurnaM, leonwehrhan, kainszs, orionarcher, 
          yuxuanzhuang, PythonFZ, laksh-krishna-sharma, orbeckst, MattTDavies,
-         talagayev, aya9aladdin
+         talagayev, aya9aladdin, ChiahsinChu
```
Comment (orbeckst) [inline package/CHANGELOG]: Please merge the current develop branch into your PR and then move all the CHANGELOG information to the 2.9.0 entry.

File: package/CHANGELOG [hunk 2]
```diff
@@ -63,6 +63,7 @@ Fixes
  * Fix groups.py doctests using sphinx directives (Issue #3925, PR #4374)
 
 Enhancements
+ * Added `precision` for XYZWriter (PR #4771)
```
Comment (orbeckst) [inline package/CHANGELOG]: move to 2.9.0

Comment (orbeckst) [reply package/CHANGELOG]: reference issue #4775 (issue numbers are more important than PRs)

File: package/MDAnalysis/coordinates/XYZ.py [hunk 3]
```diff
@@ -161,6 +168,8 @@ def __init__(self, filename, n_atoms=None, convert_units=True,
         remark: str (optional)
             single line of text ("molecule name"). By default writes MDAnalysis
             version and frame
+        precision: int (optional)
+            set precision of saved trjactory to this number of decimal places.
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/XYZ.py:172]: add a `.. versionadded:: 2.9.0` for the new kwarg
```suggestion
            set precision of saved trjactory to this number of decimal places.

            .. versionadded:: 2.9.0
```

**Review by orbeckst — CHANGES_REQUESTED** [review]
Overall the PR looks sensible to me.

Can you please cleanup CHANGELOG — we released 2.9.0 a week ago so you have to merge develop into your branch and then add your change notes under 2.9.0, please.

File: package/CHANGELOG [hunk 4]
```diff
@@ -14,13 +14,14 @@ The rules for this file:
...
 -------------------------------------------------------------------------------
-??/??/?? IAlibay
+??/??/?? IAlibay, ChiahsinChu
 
  * 2.9.0
 
 Fixes
 
 Enhancements
+  * Added `precision` for XYZWriter (Iusse #4775, PR #4771)
```
Comment (orbeckst) [inline package/CHANGELOG]: typo
```suggestion
  * Added `precision` for XYZWriter (Issue #4775, PR #4771)
```

**Review by orbeckst — APPROVED** [review]
Please just fix the small typo in CHANGELOG, otherwise looks good to me.

Thank you for your contribution!!

**Comment by orbeckst** [comment]
(The test failures are not due to your PR.)

**Review by RMeli — APPROVED** [review]

**Comment by RMeli** [comment]
Thank you @ChiahsinChu. Only RDKit tests are failing, which is a known issue we are working on, and unrelated to this PR.

**Pull request merged**